src/shared/hooks/island.ts

```diff
@@ -152,33 +152,46 @@ export function useStakeCooldownPeriod() {
 export function useIslandRealmsPaths(
   pathRef: MutableRefObject<(Path | null)[]>,
   groupRef: RefObject<Group>,
-  setHover: Dispatch<SetStateAction<boolean>>
+  setFocus: Dispatch<SetStateAction<boolean>>
 ) {
-  useLayoutEffect(() => {
-    const group = groupRef.current
-
-    pathRef.current.forEach((ref) => {
-      const stage = ref?.getStage()
-      if (!stage || !ref || !group) return () => {}
-      const defaultZ = group.zIndex()
-
-      const handleHover = (evt: Konva.KonvaEventObject<MouseEvent>) => {
-        if (evt.type === "mouseenter") {
-          stage.container().style.cursor = "pointer"
-          group.zIndex(REALMS_COUNT)
-          setHover(true)
-        } else if (evt.type === "mouseleave") {
-          stage.container().style.cursor = "default"
-          group.zIndex(defaultZ)
-          setHover(false)
-        }
-      }
+  const selectedRealmId = useDappSelector(selectDisplayedRealmId)
+
+  const onMouseEnter = useCallback(
+    (e: KonvaEventObject<MouseEvent>) => {
+      const stage = e.target.getStage()
+
+      stage
+        ?.container()
+        .style.setProperty("cursor", selectedRealmId ? "default" : "pointer")
+
+      groupRef?.current?.zIndex(REALMS_COUNT)
```

Comment:
Console seems to complaining about `REALMS_COUNT` value being used as z-index here. We can change definetely change it to a smaller value

![Zrzut ekranu 2023-11-30 o 09 22 00](https://github.com/tahowallet/dapp/assets/73061939/9a2a0e14-425e-42f3-b9da-4d5bbac068f3)